packages/std/src/coins.rs

```diff
@@ -389,4 +477,28 @@ mod tests {
         assert_eq!(coins.len(), 1);
         assert_eq!(coins.amount_of("uatom").u128(), 12345);
     }
+
+    #[test]
+    fn can_iterate_owned() {
+        let coins = mock_coins();
+        let mut moved = Coins::default();
+        for c in coins {
+            moved.add(c).unwrap();
+        }
+        assert_eq!(moved.len(), 3);
+
+        assert!(mock_coins().into_iter().eq(mock_coins().to_vec()));
+    }
+
+    #[test]
+    fn can_iterate_borrowed() {
+        let coins = mock_coins();
+        assert!(coins
+            .iter()
+            .map(|c| c.denom)
+            .eq(coins.to_vec().iter().map(|c| &c.denom)));
```

Comment:
Oh, cool, I did not know eq.

But we see an issue here. You can't compare the elements directly (&Coin and CoinRef). So is we follow this path we have to implement all the math an comparison between all combinations of Coin, &Coin and CoinRef.

Reply:
Yes, that is indeed the big problem of this approach.